libcudacxx/test/libcudacxx/cuda/proclaim_return_type.pass.cpp

```diff
@@ -93,8 +95,22 @@ int main(int argc, char ** argv)
   auto f = cuda::proclaim_return_type<bool>([] __device__() { return false; });
   auto g = cuda::proclaim_return_type<bool>([f] __device__() { return f(); });
 
+  #if !defined(TEST_COMPILER_CUDACC_BELOW_12_3) && TEST_STD_VER >= 2017
+    // Ensure type can be queried from cuda::std::invoke_result_t 
+    auto d_lm = [] __device__ () -> float { return 3.14f; };
+    auto hd_lm = [] __host__ __device__ () -> float { return 3.14f; };
+    using Td = decltype(d_lm);
+    using Thd = decltype(hd_lm);
+
+    ASSERT_SAME_TYPE(cuda::std::invoke_result_t<Td>, float);
+    ASSERT_SAME_TYPE(cuda::std::invoke_result_t<Thd>, float);
+    unused(d_lm);
+    unused(hd_lm);
```

Comment:
Suggest: Maybe I'm just being paranoid, but I think we could expand this a bit to include other return types, especially something that is a custom type whose size is larger than 4B.

```suggestion

template <typename ReturnT>
void test_lambda_return_type(){
    // Ensure type can be queried from cuda::std::invoke_result_t 
    auto d_lm = [] __device__ () -> ReturnT { return T{}; };
    auto hd_lm = [] __host__ __device__ () -> ReturnT { return T{}; };
    using Td = decltype(d_lm);
    using Thd = decltype(hd_lm);

    ASSERT_SAME_TYPE(cuda::std::invoke_result_t<Td>, ReturnT);
    ASSERT_SAME_TYPE(cuda::std::invoke_result_t<Thd>, ReturnT);
    unused(d_lm);
    unused(hd_lm);
 }
...

  struct custom_type{
     char arr[16];
  };
  
  test_lambda_return_type<float>();
  test_lambda_return_type<custom_type>();
  
```

Reply:
Fixed